api/src/main/java/org/apache/iceberg/types/PruneColumns.java

```diff
@@ -22,12 +22,14 @@
 import java.util.List;
 import java.util.Set;
 import org.apache.iceberg.Schema;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 class PruneColumns extends TypeUtil.SchemaVisitor<Type> {
   private final Set<Integer> selected;
 
   PruneColumns(Set<Integer> selected) {
+    Preconditions.checkNotNull(selected, "Selected cannot be null");
```

Comment:
Rather than just using the name of the variable, I think this should give more context to users when it happens. "Field id set" or "Selected field ids" would be better.

Reply:
I agree with you on this. I was following what I thought was somewhat of a convention - what I should really refer to as something I've seen previously in the code base - of using the argument name directly. However, I also just today saw some instances that are definitely much more explicit.

I will update to be `Selected field ids` as its inline with our definitions of `Select` and `Project` as well as it still contains the argument name in it. Please correct me if I'm incorrect about `select` vs `project`. @rdblue